Rover/fence.cpp

```diff
@@ -4,17 +4,16 @@
 void Rover::fence_check()
 {
 #if AP_FENCE_ENABLED
-    uint8_t new_breaches;  // the type of fence that has been breached
-    const uint8_t orig_breaches = fence.get_breaches();
-
-    // check for a breach
-    new_breaches = fence.check();
-
     // return immediately if motors are not armed
     if (!arming.is_armed()) {
         return;
     }
 
+    const uint8_t orig_breaches = fence.get_breaches();
+
+    // check for a breach
+    uint8_t new_breaches = fence.check();
```

Comment:
Think this is the main "update" function for fence. So even thought the return is not used we still need to call it. Otherwise you the fence arming check might not work for example.

Reply:
A fence_check is performed in the arming.pre_arm_checks method.

https://github.com/ArduPilot/ardupilot/blob/master/Rover/Rover.cpp#L450

Reply:
Right, but it does not actually check if the vehicle is inside the fence, it just uses the result of the last check. Except there now will be no result because your not running the check.

Reply:
I understood.